pydicom/valuerep.py

```diff
@@ -924,13 +944,17 @@ def _create_dict(self) -> Dict[str, str]:
         return {c: getattr(self, c, '') for c in parts}
 
     @property
-    def components(self) -> List[str]:
-        """Returns up to three decoded person name components.
+    def components(self) -> Tuple[str, ...]:
```

Comment:
The components should be immutable -> all changes should flow from changes to the encoded value.